### PR TITLE
JAMES-2675 Index all messages from a range, not only the first one

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[0c2b233b6a52a736501623c02cc038cd472ead08] [ERROR] Failures: 
[0c2b233b6a52a736501623c02cc038cd472ead08] [ERROR]   RabbitMQIMAPInconsistenciesTest 
[0c2b233b6a52a736501623c02cc038cd472ead08] Expected size:<2> but was:<0> in:
```

might be a dumb question, Is they any reason to user cucumber instead of normal integration tests setup like JMAP?

**Comment by rouazana** [comment]
> might be a dumb question, Is they any reason to user cucumber instead of normal integration tests setup like JMAP?

I found it more readable for this kind of tests. But anyway I could need to come back to Java integration tests to fix this error. Or to find a reliable way to do a rabbitMQ.await(), but I did not find any for now.

**Comment by rouazana** [comment]
> A unit test might be cool too

This is a nightmare, involving heavy mocking, so not proving so much... I will try again tomorrow, else I propose we merge it as is to be able to release it soon.

**Comment by chibenwa** [comment]
>>  A unit test might be cool too

> This is a nightmare, involving heavy mocking, so not proving so much... I will try again tomorrow, else I propose we merge it as is to be able to release it soon.

Can I get a try?

**Comment by rouazana** [comment]
thanks :+1:

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java [hunk 1]
```diff
@@ -1452,6 +1454,20 @@ public void searchShouldRetrieveExactlyMatchingMimeMessageID() throws Exception
...
+    public void copiedMessageShouldAllBeIndexed() throws Exception {
+        MailboxPath newBoxPath = MailboxPath.forUser(USERNAME, "newBox");
+        MailboxId newBoxId = storeMailboxManager.createMailbox(newBoxPath, session).get();
+
+        storeMailboxManager.copyMessages(MessageRange.all(), inboxMessageManager.getId(), newBoxId, session);
+
+        SearchQuery searchQuery = new SearchQuery();
+
+        StoreMessageManager newBox = (StoreMessageManager) storeMailboxManager.getMailbox(newBoxId, session);
+        assertThat(messageSearchIndex.search(session, newBox.getMailboxEntity(), searchQuery))
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: how do you ensure indexing is done when you run that query?

Comment (rouazana) [reply mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: I think I fixed your concern while trying to fix this test

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java [hunk 2]
```diff
@@ -1464,8 +1468,15 @@ public void copiedMessageShouldAllBeIndexed() throws Exception {
         SearchQuery searchQuery = new SearchQuery();
 
         StoreMessageManager newBox = (StoreMessageManager) storeMailboxManager.getMailbox(newBoxId, session);
-        assertThat(messageSearchIndex.search(session, newBox.getMailboxEntity(), searchQuery))
-            .hasSize(9);
+
+        Awaitility.with()
+            .pollInterval(Duration.ONE_HUNDRED_MILLISECONDS)
+            .and().with()
+            .pollDelay(new Duration(1, TimeUnit.MILLISECONDS))
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: why do you require a delay at all? 1 ms is so small that it should not make a difference, is it?

Comment (rouazana) [reply mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java]: to ensure the test is quickly run, else it waits for the poll interval (I just took the same configuration than for JMAP integration tests)

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**